wifimgr.py

```diff
@@ -281,18 +291,31 @@ def start(port=80):
         print('client connected from', addr)
         try:
             client.settimeout(5.0)
-
-            request = b""
+            request = bytearray(b"")
             try:
                 while "\r\n\r\n" not in request:
-                    request += client.recv(512)
+                    request.extend(client.recv(512))
             except OSError:
                 pass
 
-            print("Request is: {}".format(request))
-            if "HTTP" not in request:  # skip invalid requests
+            if "HTTP" not in request:
+                # skip invalid requests
                 continue
 
+            if "POST" in request and "Content-Length: " in request:
+                content_length = int(ure.search("Content-Length: ([0-9]+)?", bytes(request)).group(1))
+                content = request[bytes(request).index(b"\r\n\r\n") + 4:]
+                content_length_remaining = content_length - len(content)
+
+                while content_length_remaining > 0:
+                    chunk = client.recv(512)
+                    request.extend(chunk)
```

Comment:
as you already have the starting of the content in `content`, wouldn't you rather like to append the rest of the content to that instead of to `request`?

Reply:
or at least remember that position of \r\n\r\n (end of headers) for later, so the body can be extracted more easily.